predictors.go

```diff
@@ -0,0 +1,240 @@
+package kong
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/posener/complete"
+)
+
+// Predictor implements a predict method, in which given command line arguments returns a list of options it predicts.
+// Taken from github.com/posener/complete
+type Predictor interface {
+	Predict(PredictorArgs) []string
+}
+
+// PredictorArgs describes command line arguments used by a Predictor to predict options.
+// Taken from Args in github.com/posener/complete
+type PredictorArgs struct {
+	// All lists of all arguments in command line (not including the command itself)
+	All []string
+	// Completed lists of all completed arguments in command line,
+	// If the last one is still being typed - no space after it,
+	// it won't appear in this list of arguments.
+	Completed []string
+	// Last argument in command line, the one being typed, if the last
+	// character in the command line is a space, this argument will be empty,
+	// otherwise this would be the last word.
+	Last string
+	// LastCompleted is the last argument that was fully typed.
+	// If the last character in the command line is space, this would be the
+	// last word, otherwise, it would be the word before that.
+	LastCompleted string
+}
```

Comment:
Does this capture intra-token completions? eg. `token` where the cursor is at `tok><en`?

Reply:
Yes.  If you had typed `cmdname this is my tok<tab>`, PredictorArgs would look like this:

```go
PredictorArgs{
	All:           []string{"this", "is", "my", "tok"},
	Completed:     []string{"this", "is", "my"},
	Last:          "tok",
	LastCompleted: "my",
}
```

Likewise, if you had typed `cmdname this is my token t<4 x back arrow><tab>` (putting the cursor after the k in token), PredictorArgs would look the same as above.